src/main/scala/temple/Main.scala

```diff
@@ -1,10 +1,18 @@
 package temple
 
+import java.nio.file.FileAlreadyExistsException
+
 import temple.DSL.DSLProcessor
 import temple.utils.FileUtils
 
 /** Main entry point into the application */
 object Main extends App {
+
+  def fail(msg: String): Nothing = {
```

Comment:
Seeing as this is quite similar to a lot of failure stuff Charlie used in #22, would it maybe be worth abstracting this out to a `FailUtils`or something?

Reply:
They're a little different - Charlie's throws an exception: https://github.com/TempleEight/temple/blob/3e3f12e87a48b91f85189a6b87c375af47189d5d/src/main/scala/temple/DSL/semantics/ErrorHandling.scala#L19

Might still be worth pulling it out though